From c34b21cff262a84cafd83910a04deb7b4b3ecc57 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Sat, 19 Nov 2011 18:58:12 +0800
Subject: [PATCH 175/280] ENGR00162466 vpu: Use the buffer in
 SET_ROTATOR_OUTPUT cmd for MJPG decoder

Use the buffer in SET_ROTATOR_OUTPUT cmd for MJPG decoding no
matter there is rotation or not on mx6q. Then the API is align
with mx5 mjpg usage.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 29 ++++++++---------------------
 1 file changed, 8 insertions(+), 21 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 38d0c27..c450dc2 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -3440,27 +3440,14 @@ RetCode vpu_DecStartOneFrame(DecHandle handle, DecParam * param)
 		while (!val)
 			val = VpuReadReg(GDI_STATUS);
 
-		if (rotMir & 0x10) {
-			VpuWriteReg(GDI_INFO_CONTROL, ((pDecInfo->jpgInfo.format & 0x07) << 17) |
-							pDecInfo->rotatorStride);
-			VpuWriteReg(GDI_INFO_PIC_SIZE, (pDecInfo->jpgInfo.alignedWidth << 16) |
-							pDecInfo->jpgInfo.alignedHeight);
-			VpuWriteReg(GDI_INFO_BASE_Y,  pDecInfo->rotatorOutput.bufY);
-			VpuWriteReg(GDI_INFO_BASE_CB,  pDecInfo->rotatorOutput.bufCb);
-			VpuWriteReg(GDI_INFO_BASE_CR,  pDecInfo->rotatorOutput.bufCr);
-			VpuWriteReg(MJPEG_DPB_BASE00_REG, 0);
-		} else {
-			val = (pDecInfo->jpgInfo.frameIdx % pDecInfo->numFrameBuffers);
-			VpuWriteReg(GDI_INFO_CONTROL, ((pDecInfo->jpgInfo.format & 0x07) << 17) |
-							pDecInfo->stride);
-			VpuWriteReg(GDI_INFO_PIC_SIZE, (pDecInfo->jpgInfo.alignedWidth << 16) |
-							pDecInfo->jpgInfo.alignedHeight);
-
-			VpuWriteReg(GDI_INFO_BASE_Y,  pDecInfo->frameBufPool[val].bufY);
-			VpuWriteReg(GDI_INFO_BASE_CB,  pDecInfo->frameBufPool[val].bufCb);
-			VpuWriteReg(GDI_INFO_BASE_CR,  pDecInfo->frameBufPool[val].bufCr);
-			VpuWriteReg(MJPEG_DPB_BASE00_REG, 0);
-		}
+		VpuWriteReg(GDI_INFO_CONTROL, ((pDecInfo->jpgInfo.format & 0x07) << 17) |
+						pDecInfo->rotatorStride);
+		VpuWriteReg(GDI_INFO_PIC_SIZE, (pDecInfo->jpgInfo.alignedWidth << 16) |
+						pDecInfo->jpgInfo.alignedHeight);
+		VpuWriteReg(GDI_INFO_BASE_Y,  pDecInfo->rotatorOutput.bufY);
+		VpuWriteReg(GDI_INFO_BASE_CB,  pDecInfo->rotatorOutput.bufCb);
+		VpuWriteReg(GDI_INFO_BASE_CR,  pDecInfo->rotatorOutput.bufCr);
+		VpuWriteReg(MJPEG_DPB_BASE00_REG, 0);
 
 		VpuWriteReg(GDI_CONTROL, 0);
 		VpuWriteReg(GDI_PIC_INIT_HOST, 1);
-- 
1.8.0

